### PR TITLE
Fix tests with node 15

### PR DESCRIPTION
**Pull request by Chocobozzz**
Please after the merge, could you create a patch release?

It blocks node upgrade of projects that uses your library: https://github.com/Chocobozzz/PeerTube/issues/2700

Thanks for your work :)

### CODE REVIEW

**Comment by rimiti** [comment]
@kwhitley Is it possible to merge this PR?

**Comment by kontrollanten** [comment]
@Chocobozzz Maybe we should remove node 6 support since it's breaking this PR? I think that all support for node <9 [should be removed,](https://en.wikipedia.org/wiki/Node.js#Releases) but I guess that's out of scope for this PR :)

**Comment by rimiti** [comment]
@kontrollanten Agreed with you, node v6 could be serenely removed.

Like some other users (I guess) I'd like to use the latest Node LTS version, is it possible for you @kwhitley to push -up this urgent topic?

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 language: node_js
 node_js:
-  - 6.0.0
-  - 7.0.0
-  - 8.0.0
-  - 10.0.0
-  - 11.0.0
-  - 12.0.0
+  - 6
```
Comment (kontrollanten) [inline .travis.yml:3]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "prepublish": "npm run test"
   },
   "engines": {
-    "node": ">=6.2.0 <13"
+    "node": ">=6.2.0 <=15"
```
Comment (kontrollanten) [inline package.json:13]:
```suggestion
    "node": ">=8.0.0 <=15"
```

**Review by kontrollanten — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-  - 7.0.0
-  - 8.0.0
-  - 10.0.0
-  - 11.0.0
-  - 12.0.0
+  - 8
+  - 10
+  - 11
+  - 12
+  - 14
```
Comment (rimiti) [inline .travis.yml:7]: 👍

**Comment by kontrollanten** [comment]
As far as I understand the failing tests is due to a race condition in `clearing cache cancels expiration callback`, both trace backs are pointing to that tests. But I've to insight in how apicache works to understand what's supposed to be tested; by the name it looks like it's testing that the expiration callback isn't called, but by the code it looks like something else. Someone with more knowledge are welcome to help :)

**Comment by Chocobozzz** [comment]
I fixed the expiration tests

**Comment by kwhitley** [comment]
Thanks for the fix @Chocobozzz!  I got to the failing point yesterday and had to shelve it again...

I'm going to release this (sadly) as a major version, as it absolutely breaks backwards compat (by dropping old Node versions).  The only question is whether to even try to maintain restify support within this new scope.  If no one uses it, better to drop support along with the old Node versions to simplify the codebase and make future support less painful.

Thoughts?

**Comment by kwhitley** [comment]
Getting this error with the branch:
image

Then under restify as well:
image

image

**Comment by rotsee** [comment]
> The only question is whether to even try to maintain restify support within this new scope.

FWIW I use this with Restify in production, and would hate to see that support dropped... :/

**Comment by Chocobozzz** [comment]
@kwhitley I don't have these errors, and neither does travis. What is your nodejs version? 

This PR was just fixing tests and add node 15 support, not removing node 6 support. I think we should just merge the test fixes, support node 15 and keep compat' with restify and node 6. They worked in 1.5.2 so I don't see why they would not work anymore now.

**Comment by rimiti** [comment]
Agreed with what you said @Chocobozzz.

**Comment by Chocobozzz** [comment]
What do you think @kwhitley?

**Review by rimiti — APPROVED** [review]
LGTM

**Comment by vikasg603** [comment]
Any update for the new release?
I am using Node 14, and I am continuously getting the warning from npm.

**Comment by kwhitley** [comment]
Merged and going out tonight... thanks so much @Chocobozzz and everyone else for your patience! <3

Sincerest apologies for how ridiculously long this has taken to push through... I've been swamped with a few other libraries and my upcoming Cloudflare Workers work (on top of full time job)!

Cheers all and thanks again!

**Comment by Chocobozzz** [comment]
No problem thanks for you work and your lib @kwhitley :kissing_cat:

**Pull request merged**